buildSrc/src/main/kotlin/com/datadog/gradle/plugin/benchmark/BenchmarkJsonFileVisitor.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-2020 Datadog, Inc.
+ */
+
+package com.datadog.gradle.plugin.benchmark
+
+import com.datadog.gradle.plugin.benchmark.model.Benchmark
+import com.datadog.gradle.plugin.benchmark.model.BenchmarkResult
+import com.google.gson.Gson
+import java.io.File
+
+class BenchmarkJsonFileVisitor {
+
+    // region ReviewJsonFileVisitor
+
+    fun visitBenchmarkJsonFile(jsonFile: File, thresholds: Map<String, Long>) {
+        val gson = Gson()
+        val benchmarkResult = gson.fromJson(jsonFile.readText(), BenchmarkResult::class.java)
+
+        println(benchmarkResult.context.targetBuild)
+
+        benchmarkResult.benchmarks.forEach {
+            processBenchmarkResults(it, thresholds)
+        }
+    }
+
+    // endregion
+
+    // region Internal
+
+    private fun processBenchmarkResults(
+        benchmark: Benchmark,
+        thresholds: Map<String, Long>
+    ) {
+        val testName = if (benchmark.name.startsWith("EMULATOR_")) {
+            benchmark.name.substringAfter('_')
+        } else {
+            benchmark.name
+        }
+        val expectedThreshold = thresholds[testName]
+        if (expectedThreshold == null) {
+            throw IllegalStateException(
```

Comment:
I am thinking is too drastic to throw an exception here....maybe just a log will be enough.

Reply:
Well, the goal is to know when a benchmark reports performance issues. If we just logged things, will we really go to Bitrise to read the logs ? Making the build break feals safer.